api/src/test/java/org/pmiops/workbench/api/DisksControllerTest.java

```diff
@@ -0,0 +1,236 @@
+
+@DataJpaTest
+public class DisksControllerTest {
+  private static final String WORKSPACE_NS = "workspace-ns";
+  private static final String GOOGLE_PROJECT_ID = "aou-gcp-id";
+  private static final String WORKSPACE_ID = "myfirstworkspace";
+  private static final String WORKSPACE_NAME = "My First Workspace";
+  private static final String LOGGED_IN_USER_EMAIL = "bob@gmail.com";
+  private static final Instant NOW = Instant.now();
+
+  private static WorkbenchConfig config = new WorkbenchConfig();
+  private static DbUser user = new DbUser();
+
+  @TestConfiguration
+  @Import({
+    DisksController.class,
+    FakeClockConfiguration.class,
+    LeonardoApiHelper.class,
+    LeonardoMapperImpl.class,
+  })
+  static class Configuration {
+
+    @Bean
+    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+    public WorkbenchConfig workbenchConfig() {
+      return config;
+    }
+
+    @Bean
+    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+    DbUser user() {
+      return user;
+    }
+  }
+
+  @Captor private ArgumentCaptor<LeonardoUpdateDiskRequest> updateDiskRequestCaptor;
+
+  @MockBean LeonardoApiClient mockLeonardoApiClient;
+  @MockBean WorkspaceService mockWorkspaceService;
+
+  @Autowired UserDao userDao;
+  @Autowired DisksController disksController;
+
+  @BeforeEach
+  public void setUp() {
+    config = WorkbenchConfig.createEmptyConfig();
+
+    user = new DbUser();
+    user.setUsername(LOGGED_IN_USER_EMAIL);
+    user.setUserId(123L);
+
+    DbWorkspace testWorkspace =
+        new DbWorkspace()
+            .setWorkspaceNamespace(WORKSPACE_NS)
+            .setGoogleProject(GOOGLE_PROJECT_ID)
+            .setName(WORKSPACE_NAME)
+            .setFirecloudName(WORKSPACE_ID);
+    doReturn(testWorkspace).when(mockWorkspaceService).lookupWorkspaceByNamespace((WORKSPACE_NS));
+  }
+
+  @Test
+  public void testGetPD() throws ApiException {
+    String createDate = "2021-08-06T16:57:29.827954Z";
+    String pdName = "pdName";
+    LeonardoGetPersistentDiskResponse getResponse =
+        new LeonardoGetPersistentDiskResponse()
+            .name(pdName)
+            .size(300)
+            .diskType(LeonardoDiskType.STANDARD)
+            .status(LeonardoDiskStatus.READY)
+            .auditInfo(new LeonardoAuditInfo().createdDate(createDate).creator(user.getUsername()))
+            .googleProject(GOOGLE_PROJECT_ID);
+
+    Disk disk =
+        new Disk()
+            .name(pdName)
+            .size(300)
+            .diskType(DiskType.STANDARD)
+            .status(DiskStatus.READY)
+            .createdDate(createDate)
+            .creator(user.getUsername())
+            .isGceRuntime(true);
+
+    when(mockLeonardoApiClient.getPersistentDisk(GOOGLE_PROJECT_ID, pdName))
+        .thenReturn(getResponse);
+    assertThat(disksController.getDisk(WORKSPACE_NS, pdName).getBody()).isEqualTo(disk);
+  }
+
+  @Test
+  public void testListPD() throws ApiException {
+    // RStudio Disk: 2 are active, returns the newer one.
```

Comment:
This is the most complicated part that we want to test.

Reply:
This test fails sometimes - is there something nondeteministic going on here?  https://app.circleci.com/pipelines/github/all-of-us/workbench/23011/workflows/2ee6cb7a-d876-4118-8b3c-48c693903880/jobs/247000

Reply:
See https://github.com/all-of-us/workbench/pull/7141